File: exercises/io/cs/read_file.cs
using System;
using static System.Math;

static class ReadFile{
    static int Main(string[] args){
        if(args.Length != 2) {
            Console.Error.Write("Wrong number of arguments to read file.\n");
            return 1;
            }
        var infile = new System.IO.StreamReader(args[0]);
        var outfile = new System.IO.StreamWriter(args[1]);
        string s;
        double x;
        outfile.Write("x\tcos(x)\tsin(x)\n");
        while(true){
            s = infile.ReadLine();
            if (s == null){
                break;
            }
            x = double.Parse(s);
            outfile.Write("{0}\t{1}\t{2}\n",x, Cos(x), Sin(x));
        }
        
        return 0;
    }
}

File: exercises/io/c/read_file.c
#include <stdio.h>
#include <stdlib.h>
#include "write_data.h"

int main(int argc, char const *argv[]){
    if (argc != 3){
        fprintf(stderr, "Wrong number of arguments supplied.\n");
        return -1;
    }
    double x;
    FILE *infile, *outfile;
    infile = fopen(argv[1], "r"), 
    outfile = fopen(argv[2], "w");
    fprintf(outfile, "x\tcos(x)\tsin(x)\n");
    do {
        int items=fscanf(infile, "%lg", &x);
        if(items==EOF){
            break;
        }
        write_data(x, outfile);
    } while(1);
    return 0;
}